client/src/pages/NrMap.js

```diff
@@ -99,81 +113,74 @@ class NrMap extends Component {
           "icon",
         ],
       };
-      new window.google.maps.places.PlacesService(this.googleMap).getDetails(
+
+      new window.google.maps.places.PlacesService(map.googleMap).getDetails(
         request,
         (req, status) => {
           if (status == window.google.maps.places.PlacesServiceStatus.OK) {
-            this.setState((prevState) => ({
-              plcDtl: [...prevState.plcDtl, req],
-            }));
+            setPlcDtl((prevState) => {
```

Comment:
If we change the initial state for `plcDtl ` to be `[]`, let's rewrite this to be `setPlcDtl([...plcDtl, req])`.

Reply:
`setPlcDtl([...plcDtl, req]) ` tried this way but only displays one card, it is not getting the previous updated state.

Reply:
Ok I see. I would think that `p1cD1` would be equivalent to `prevState`. So it's weird that it's not getting the previous updated state.

Reply:
Yes, I also thought that way. React and goole libraries interacting with the Dom at the same time makes the behaviour of this component is a bit tricky.